### DISCUSSION TITLE
Vscode Extension currently unmaintained

### DISCUSSION THREAD

**Discussion by Mrcubix** [AvaloniaUI/Avalonia#15350] [Q&A]
It seem that the VSCode Extension Maintainer seem to have  found an opportunity, which is a good thing. 

Unfortunately, that means they won't be able to focus on the extension, which is currently broken & unusable.

Hope someone can take over the development soon!

**Comment by MikeCodesDotNET** [comment]
Prashant's exceptional work built the VSCode extension with a notable investment from AvaloniaUI OÜ. As a small company, our resources need to be directed towards areas directly impacting our business. To provide context, this investment was equivalent to several years of donations. 

Because of that, we're transitioning it to the community to maintain (should they want to). If we see a significant increase in Github donations, we may look again at funding its further development in the medium term.

**Comment by Shadowblitz16** [comment]
Wait so you guys are dropping cross platform vscode plugin for a windows only vs plugin?

**Reply by rabbitism** [reply]
Rider is crossplatform and it behaves better for Avalonia app development

**Reply by Shadowblitz16** [reply]
> Rider is crossplatform and it behaves better for Avalonia app development

Yes but it cost alot of money and it's not just a one time price.

**Reply by rabbitism** [reply]
> > Rider is crossplatform and it behaves better for Avalonia app development
> 
> Yes but it cost alot of money and it's not just a one time price.

You can always use EAP version (if available) which is free.

**Reply by MikeCodesDotNET** [reply]
> Wait so you guys are dropping cross platform vscode plugin for a windows only vs plugin?

Not at all. VSCode is still there, and you can use JetBrains Rider. 

We have asked the community that uses the extension to contribute and maintain it. We've already invested a large amount of money in building it in the hope that the community will take it over.

**Reply by Mrcubix** [reply]
Main issue i have with rider is the performance, it's slow, cause high CPU usage, indexing happens all the time which slows the process even more.
I also find its interface bloated.
It is not an alternative in these conditions.

But anyway, is there some documentation on how the core of the previewer works and ways extensions may interact with it?

**Reply by Shadowblitz16** [reply]
> > > Rider is crossplatform and it behaves better for Avalonia app development
> > 
> > 
> > Yes but it cost alot of money and it's not just a one time price.
> 
> You can always use EAP version (if available) which is free.

if available means it's not reliable
